### PR TITLE
database configuration models sequelize

### PR DESCRIPTION
**Pull request by AbdallahAbujazar**
Relates#13

### CODE REVIEW

File: src/db/models.ts
Comment (haitham-akram) [inline src/db/models.ts:1]: You can separate every model in its own file so it's more organizable and easy to read.

File: src/app.ts
Comment (haitham-akram) [inline src/app.ts:1]: remove the `;` from your code and see if eslint work.

File: src/db/config.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { Sequelize } from 'sequelize';
+import dotenv from 'dotenv';
+
+dotenv.config();
+
+const { DB_DATABASE, DB_USERNAME, DB_PASSWORD, DB_HOST, DB_DIALECT } = process.env;
```
Comment (Mu7ammadAbed) [inline src/db/config.ts:6]: We will change the environment variables names to these names, and we will not take the dialect here
```suggestion
const { NODE_ENV, DEV_DB_URL, TEST_DB_URL, DATABASE_URL, } = process.env;
```

File: src/db/config.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+const sequelize = new Sequelize(DB_DATABASE!, DB_USERNAME!, DB_PASSWORD!, {
+  host: DB_HOST!,
+  dialect: DB_DIALECT as 'mysql' | 'postgres' | 'sqlite' | 'mariadb' | 'mssql',
+  logging: false, 
+});
```
Comment (Mu7ammadAbed) [inline src/db/config.ts:12]:
```suggestion
let url:string;
let ssl:boolean | { rejectUnauthorized: boolean } = false;

switch (NODE_ENV) {
  case 'production': url = DATABASE_URL;
    ssl = { rejectUnauthorized: false };
    break;
  case 'development': url = DEV_DB_URL;
    break;
  case 'test': url = TEST_DB_URL;
    break;
  default: throw new Error('invalid database url');
}

const sequelize = new Sequelize(url, { logging: false, dialect: 'postgres', dialectOptions: { ssl } });
```

File: src/app.ts [hunk 3]
```diff
@@ -1,30 +1,40 @@
...
+        res.sendFile(join(__dirname, '..', 'client', 'build', 'index.html'));
+      });
+    }
+    app.listen(PORT ?? 3000, () => {
+      console.log(`App live on: http://localhost:${PORT ?? 3000}`);
+    });
+  } catch (error) {
+    console.error('Error syncing database:', error);
+  }
+})();
```
Comment (Mu7ammadAbed) [inline src/app.ts:38]: restore these to what it was before and add the database sync to the `index.ts`

```suggestion
if (NODE_ENV === 'production') {
  app.use(express.static(join(__dirname, '..', 'client', 'build')))
  app.get('/*', (_req, res) => {
    res.sendFile(join(__dirname, '..', 'client', 'build', 'index.html'))
  })
}
```

**Review by Mu7ammadAbed — CHANGES_REQUESTED** [review]

**Pull request closed without merging**